src/main/java/net/dv8tion/jda/core/requests/restaction/MessageAction.java

```diff
@@ -517,6 +595,9 @@ protected RequestBody asMultipart()
         }
         if (!isEmpty())
             builder.addFormDataPart("payload_json", getJSON().toString());
+        // clear remaining resources, they will be closed after being sent
+        files.clear();
+        ownedResources.clear();
```

Comment:
not sure if this is only called by actual execution of the restaction (and if that will cause the MultipartBody to close all given streams)

Reply:
The body does close the stream, it is only called on execution.